php/tests/GeneratedClassTest.php

```diff
@@ -497,6 +497,47 @@ public function testBoolField()
         $this->assertSame(false, $m->getOptionalBool());
     }
 
+    public function testBoolFromDoubleArrayConstructor()
+    {
+        $m = new TestMessage(['optional_bool' => -0.0]);
+        $this->assertFalse($m->getOptionalBool());
```

Comment:
This fails on `main` and passes on this branch (effectively proving the effectiveness of the fix). Same goes for line 533